zapcore/console_encoder.go

```diff
@@ -125,10 +125,12 @@ func (c consoleEncoder) EncodeEntry(ent Entry, fields []Field) (*buffer.Buffer,
 		line.AppendString(ent.Stack)
 	}
 
-	if c.LineEnding != "" {
-		line.AppendString(c.LineEnding)
-	} else {
-		line.AppendString(DefaultLineEnding)
+	if !c.SkipLineEnding {
```

Comment:
It'd be preferable to change `DefaultLineEnding` to an empty string instead of doing this because this is essentially adding an additional boolean check for configs that didn't specify the SkipLineEnding every time we encode.

Reply:
Wouldn't changing the default value mean breaking backward compatibility? I suppose there are people expecting to see their logs going to a new line every time, and pulling a change like that would break that behaviour, I think.

Reply:
sorry, I should've been more clear w/ my comment - I meant that you'd just need to change DefaultLineEnding to an empty string *if* `SkipLineEnding` is set to true.

Reply:
I see. So merging with your other comment below if would be changing `SkipLineEnding ` to a var and computing the "real" `LineEnding` at `NewXEncoder` time, right?

Reply:
`SkipLineEnding` can stay in `EncoderConfig`; What I'd imagine would change is somewhere along the lines of:

```
 func newJSONEncoder(cfg EncoderConfig, spaced bool) *jsonEncoder {
+       if c.SkipLineEnding {
+               cfg.LineEnding = ""
+       } else if cfg.LineEnding == "" {
+               cfg.LineEnding = DefaultLineEnding
+       }
```

And in these lines, we'd simply need to do:
`line.AppendString(cfg.LineEnding)` instead of doing the if checks over and over.

`NewConsoleEncoder` uses `newJSONEncoder` so you wouldn't need to change anything else.

Reply:
should be handled within [the new commit](https://github.com/uber-go/zap/pull/989/commits/6fa2cff0b691b932c5b8a17f6f1f97f4a1697fdb)